TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -13,492 +13,481 @@
 #if DEBUG
-            bool debug = GlobalConfig.Instance.Debug.Switches[11];
+			bool debug = GlobalConfig.Instance.Debug.Switches[11];
 #endif
-	        if (!Options.turnOnRed)
-		        return false;
-	        
-            SegmentGeometry segGeo = SegmentGeometry.Get(segmentId);
-            SegmentEndGeometry endGeo = segGeo?.GetEnd(startNode);
-
-            bool ret = (node.m_flags & (NetNode.Flags.TrafficLights)) != NetNode.Flags.None &&
-                    node.Info?.m_class?.m_service != ItemClass.Service.Beautification &&
-                    !LogicUtil.CheckFlags((uint)node.m_flags, (uint)(NetNode.Flags.LevelCrossing)) &&
-                    !endGeo.OutgoingOneWay &&
-                    segGeo?.HasValidTurnOnRedOutgoingSegment(endGeo) == true;
+
+			SegmentGeometry segGeo = SegmentGeometry.Get(segmentId);
+			SegmentEndGeometry endGeo = segGeo?.GetEnd(startNode);
+
+			bool ret = (node.m_flags & (NetNode.Flags.TrafficLights | NetNode.Flags.LevelCrossing)) == NetNode.Flags.TrafficLights &&
+					node.Info?.m_class?.m_service != ItemClass.Service.Beautification &&
+					!endGeo.OutgoingOneWay &&
+					endGeo.NumOutgoingSegments > 1 &&
+					((Services.SimulationService.LeftHandDrive && endGeo.NumOutgoingLeftSegments > 0) ||
+					(!Services.SimulationService.LeftHandDrive && endGeo.NumOutgoingRightSegments > 0));
 
 #if DEBUG
-            if (debug)
-                Log._Debug($"JunctionRestrictionsManager.IsTurnOnRedAllowedConfigurable({segmentId}, {startNode}): ret={ret}, flags={node.m_flags}, service={node.Info?.m_class?.m_service}");
+			if (debug)
+				Log._Debug($"JunctionRestrictionsManager.IsTurnOnRedAllowedConfigurable({segmentId}, {startNode}): ret={ret}, flags={node.m_flags}, service={node.Info?.m_class?.m_service}");
 #endif
 
-            return ret;
-        }
+			return ret;
+		}
 
-        public bool GetDefaultTurnOnRedAllowed(ushort segmentId, bool startNode, ref NetNode node) {
+		public bool GetDefaultTurnOnRedAllowed(ushort segmentId, bool startNode, ref NetNode node) {
 #if DEBUG
-            bool debug = GlobalConfig.Instance.Debug.Switches[11];
+			bool debug = GlobalConfig.Instance.Debug.Switches[11];
 #endif
 
-            if (!IsTurnOnRedAllowedConfigurable(segmentId, startNode, ref node)) {
+			if (!IsTurnOnRedAllowedConfigurable(segmentId, startNode, ref node)) {
 #if DEBUG
-                if (debug)
-                    Log._Debug($"JunctionRestrictionsManager.IsTurnOnRedAllowedConfigurable({segmentId}, {startNode}): Setting is not configurable. res=false");
+				if (debug)
+					Log._Debug($"JunctionRestrictionsManager.IsTurnOnRedAllowedConfigurable({segmentId}, {startNode}): Setting is not configurable. res=false");
 #endif
-                return false;
-            }
-			// Feature enabled and depends on default behaviour setting
-            bool ret = Options.turnOnRed && Options.turnOnRedEnabledByDefault; 
```

Comment:
I unified both flags into one. The feature should not have any major impact on performance, thus a feature toggle is not required in my opinion.

Reply:
I think that will change behavior now, because that flag was used for switching from enable at all junction(disable on selected) and disable at all(enable at selected). I am not sure you know what I mean.

Reply:
That's what the "new" flag `allowTurnOnRed` does.
So, `allowTurnOnRed` = `turnOnRedEnabledByDefault` and
`turnOnRed` was solely a feature toggle which I removed.

Reply:
Of course, I could add the feature toggle but it confused me while reading through the code. xD

Reply:
Oh I see it now

Reply:
I restored the feature toggle but it does not need to be checked here. If `Options.turnOnRedEnabled = false` it is guaranteed that `Options.allowTurnOnRed` is also `false`.